### PR TITLE
Improve location elapsed time difference calculation

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
Use Kotlin's `Duration` class to improve the location time difference calculations.

### CODE REVIEW

**Comment by westnordost** [comment]
Can you explain in what way this improves the location time difference calculations?

**Comment by Isira-Seneviratne** [comment]
> Can you explain in what way this improves the location time difference calculations?

The duration class automatically handles the time unit conversions, so the code can be made a bit easier to understand.

File: app/src/main/java/de/westnordost/streetcomplete/util/ktx/Location.kt [hunk 1]
```diff
@@ -2,5 +2,8 @@ package de.westnordost.streetcomplete.util.ktx
 
 import android.location.Location
 import de.westnordost.streetcomplete.data.osm.mapdata.LatLon
+import kotlin.time.Duration.Companion.nanoseconds
 
 fun Location.toLatLon() = LatLon(latitude, longitude)
+
+val Location.elapsedDuration get() = elapsedRealtimeNanos.nanoseconds
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/util/ktx/Location.kt:9]: Can you add the return type here?

File: app/src/main/java/de/westnordost/streetcomplete/util/location/FineLocationManager.kt [hunk 2]
```diff
@@ -86,34 +87,30 @@ class FineLocationManager(context: Context, locationUpdateCallback: (Location) -
...
-    // the use of nanoseconds is necessary because it is the only way to get
-    // elapsedRealtime of a location before API level 33
-    val timeDelta = this.elapsedRealtimeNanos - previous.elapsedRealtimeNanos
-    val isMuchNewer = timeDelta > TWO_MINUTES_IN_NANOSECONDS
-    val isMuchOlder = timeDelta < -TWO_MINUTES_IN_NANOSECONDS
-    val isNewer = timeDelta > 0L
+    val diffMinutes = (elapsedDuration - previous.elapsedDuration).inWholeMinutes
+    val isMuchNewer = diffMinutes > 2
+    val isMuchOlder = diffMinutes < -2
+    val isNewer = diffMinutes > 0L
```
[the remark this answers is not in the document]
Comment (Isira-Seneviratne) [reply app/src/main/java/de/westnordost/streetcomplete/util/location/FineLocationManager.kt:105]: Good catch, will remove it.

**Comment by Helium314** [comment]
This introduces rather bad location update behavior, see https://github.com/Helium314/SCEE/issues/524

I suspect the issue is
```kotlin
    val diffMinutes = (elapsedDuration - previous.elapsedDuration).inWholeMinutes
    val isMuchNewer = diffMinutes > 2
    val isMuchOlder = diffMinutes < -2
    val isNewer = diffMinutes > 0
```
Comparing only whole minutes might require full 60s difference for a location to count as newer.

**Comment by westnordost** [comment]
Hmm, do you have an idea how to fix that?

**Comment by Helium314** [comment]
I didn't test whether it really is this part of the code (would need to go outside again).
But if it is, a simple fix would be using seconds instead of minutes.

**Comment by westnordost** [comment]
Okay, that should do it.

**Pull request merged**